### PR TITLE
Krige, Field and Covmodel update

### PR DESCRIPTION
**Pull request by MuellerSeb**
With this PR a revised kriging submodule is proposed. Some minor additions to Covmodel and Field were also made in the course of this.

# Kriging changes

Now the following kriging methods are provided, based on a new kriging base class:

- Simple
- Ordinary
- Universal
- External Drift Kriging
- Detrended Kriging

All methods can make use of a callable trend like in detrended kriging (which is simply a shortcut for simple kriging with a demanded trend).

# Field changes

- added ``pre_pos`` method to Field class to pre-process the position tuple
- conditioned SRF now check if the field is "scalar"

# CovModel changes

- in 3D the input of anisotropy is now treated slightly different:
  - single given anisotropy value [e] is converted to [1, e] (it was [e, e] before)
  - two given length-scales [l_1, l_2] are converted to [l_1, l_2, l_2] (it was [l_1, l_2, l_1] before)
- the TPLGaussian and TPLExponential now have analytical spectra
- added property ``is_isotropic`` to CovModel

# Still missing

- [x] tests
- [x] new style examples

### CODE REVIEW

**Comment by MuellerSeb** [comment]
Somehow I polluted the history by rebasing locally.

**Comment by LSchueler** [comment]
I'm your Mr. Wolf for git problems right?! :-)

I'll see what I can do.

**Comment by LSchueler** [comment]
Did I get it correctly, that all commits from b13845b8141e13275774efe8b70057e621ea39a1 to 87bb37d65eb2388b7aec06fb914407f714b81fe2 can be deleted? And the merge c14ab5be4399df63e03aaaed46c63ec85118ae44 too?

**Comment by MuellerSeb** [comment]
@LSchueler : yepp.

**Comment by LSchueler** [comment]
Code you check my rebase?

**Comment by MuellerSeb** [comment]
I think, I will add some tests here.

**Comment by MuellerSeb** [comment]
I'd say it's ready for review. @LSchueler

**Comment by MuellerSeb** [comment]
Resolved merge-conflicts.

File: gstools/field/base.py [hunk 1]
```diff
@@ -146,6 +154,58 @@ def mesh(
...
+        x, y, z = pos2xyz(pos, max_dim=self.model.dim)
+        pos = xyz2pos(x, y, z)
+        mesh_type_gen = mesh_type
+        # format the positional arguments of the mesh
+        check_mesh(self.model.dim, x, y, z, mesh_type)
+        mesh_type_changed = False
+        axis_lens = None
+        if (
+            self.model.do_rotation or make_unstruct
+        ) and mesh_type == "structured":
```
Comment (LSchueler) [inline gstools/field/base.py]: This is very unreadable. I suggest a line break before or after `and`.

Comment (MuellerSeb) [reply gstools/field/base.py]: The holy black said it is ok so. Ahmen.

**Comment by MuellerSeb** [comment]
> Great work!
> 
> Just some very minor things as comments in the code and sometimes you use the variable `ax_lens` and sometimes `axis_lens`. We should stick to only one.
> 
> And I pushed some typo fixes.

Thanks for the typo hunting! I renamed every occurrence `axis_lens` so its unique.

**Pull request merged**